### PR TITLE
Allow VectorDataset to accept list of files

### PR DESCRIPTION
**Pull request by adriantre**
Extend on #1442 to implement changes for VectorDataset

NB: Change base branch of this PR when #1442 is merged.
Any feedback on best practices / better approaches for git in this regard is much appreciated.

### CODE REVIEW

File: torchgeo/datasets/geo.py [hunk 1]
```diff
@@ -75,6 +75,13 @@ class GeoDataset(Dataset[dict[str, Any]], abc.ABC):
     _crs = CRS.from_epsg(4326)
     _res = 0.0
 
+    #: Glob expression used to search for files.
+    #:
+    #: This expression should be specific enough that it will not pick up files from
+    #: other datasets. It should not include a file extension, as the dataset may be in
+    #: a different file format than what it was originally downloaded as.
+    filename_glob = "*"
```
Comment (adriantre) [inline torchgeo/datasets/geo.py:83]: This now need to be defined here in GeoDataset although it is only used in Raster and Vector

Comment (adamjstewart) [reply torchgeo/datasets/geo.py:83]: Yes, that's fine.

File: torchgeo/datasets/landcoverai.py [hunk 2]
```diff
@@ -87,6 +90,10 @@ def __init__(
             RuntimeError: if ``download=False`` and data is not found, or checksums
                 don't match
         """
+        if isinstance(root, list):
+            # TODO: Workaround until list are fully implemented
+            root = root[0]
```
Comment (adriantre) [inline torchgeo/datasets/landcoverai.py:95]: Could add `assert isinstance(self.paths, str)` instead. Codecov complains

File: torchgeo/datasets/chesapeake.py [hunk 3]
```diff
@@ -704,7 +704,7 @@ def exists(filename: str) -> bool:
             return os.path.exists(os.path.join(self.root, filename))
 
         # Check if the extracted files already exist
-        if all(map(exists, self.files)):
+        if all(map(exists, self._files)):
             return
```
Comment (adriantre) [inline torchgeo/datasets/chesapeake.py:708]: Codecov says this is not covered.

File: torchgeo/datasets/geo.py [hunk 4]
```diff
@@ -269,6 +279,32 @@ def res(self, new_res: float) -> None:
...
+            paths = self.paths
+
+        # Using set to remove any duplicates if directories are overlapping
+        files: set[str] = set()
+        for path in paths:
+            if os.path.isdir(path):
+                pathname = os.path.join(path, "**", self.filename_glob)
+                files |= set(glob.iglob(pathname, recursive=True))
+            else:
+                files.add(path)
```
Comment (adriantre) [inline torchgeo/datasets/geo.py:304]: Codecov says this is not covered.

File: torchgeo/datasets/geo.py [hunk 5]
```diff
@@ -269,6 +279,32 @@ def res(self, new_res: float) -> None:
...
+        Returns:
+            All files in the dataset.
+
+        .. versionadded:: 0.5
+        """
+        # Make iterable
+        if isinstance(self.paths, str):
+            paths: Iterable[str] = [self.paths]
+        else:
+            paths = self.paths
```
Comment (adriantre) [inline torchgeo/datasets/geo.py:295]: Codecov says this is not covered.

File: torchgeo/datasets/chesapeake.py [hunk 6]
```diff
@@ -495,7 +495,7 @@ class ChesapeakeCVPR(GeoDataset):
     )
 
     # these are used to check the integrity of the dataset
-    files = [
+    _files = [
```
Comment (adriantre) [inline torchgeo/datasets/chesapeake.py:498]: This collides with the `@property` files.

**Review by adamjstewart — APPROVED** [review]

**Pull request merged**